restaurant_app/script.js

```diff
@@ -0,0 +1,324 @@
+  item13: { name: "Strawberry", cost: 170, category: "Ice-cream" },
+};
+
+var tables = {
+  table1: { cost: 0, items: 0, orders: {} },
+  table2: { cost: 0, items: 0, orders: {} },
+  table3: { cost: 0, items: 0, orders: {} },
+  table4: { cost: 0, items: 0, orders: {} },
+};
+const len = Object.keys(tables).length;
+const menuLen = Object.keys(menu).length;
+var tableContent = document.getElementById("tables");
+var menuContent = document.getElementById("menu");
+displayTables();
+displayMenu();
+function displayTables() {
+  for (let i = 1; i <= len; i++) {
+    let { cost, items } = tables["table" + i];
+
+    let tableItem = document.createElement("li");
+    tableItem.className = "table";
+    tableItem.id = `table${i}`;
+    tableItem.onclick = function () {
+      openPopUp(`table-${i}`);
+    };
+    tableItem.ondrop = function (event) {
+      drop(event, "table" + i);
+    };
+    tableItem.ondragover = function (event) {
+      allowDrop(event);
+    };
+
+    let h2 = document.createElement("h2");
+    h2.textContent = "Table-" + i;
+
+    let p = document.createElement("p");
+    p.textContent = " Items : " + items + " | Total : " + cost.toFixed(2);
+
+    tableItem.appendChild(h2);
+    tableItem.appendChild(p);
+
+    tableContent.appendChild(tableItem);
+  }
+}
+
+function displayMenu() {
+  const menuByCategory = new Map();
+  for (let i = 1; i <= menuLen; i++) {
+    const { name, cost, category } = menu["item" + i];
+    if (!menuByCategory.has(category)) {
+      menuByCategory.set(category, []);
+    }
+    menuByCategory.get(category).push({ name, cost });
+  }
+  let i = 1;
+
+  function displayCategoryItems(category, items) {
+    const categoryHeader = document.createElement("h2");
+    categoryHeader.id = "category";
+    categoryHeader.textContent = category;
+    menuContent.appendChild(categoryHeader);
+
+    for (const item of items) {
+      const li = document.createElement("li");
+      li.id = "item" + i;
+      li.setAttribute("category", category);
+      li.draggable = true;
+      li.ondragstart = function (event) {
+        drag(event);
+      };
+
+      const h2 = document.createElement("h2");
+      h2.textContent = item.name;
+      li.appendChild(h2);
+
+      const costP = document.createElement("p");
+      costP.textContent = item.cost + ".00";
+      li.appendChild(costP);
+
+      menuContent.appendChild(li);
+      i++;
+    }
+  }
+
+  for (const [category, items] of menuByCategory.entries()) {
+    displayCategoryItems(category, items);
+  }
+}
+
+function filterTables() {
+  const input = document.getElementById("searchTable").value.toLowerCase();
+  const str = input.split("-");
+  const st = str[0] + str[1];
+  if (tables.hasOwnProperty(st)) {
+    makeTableVisible(st);
+  } else {
+    tableContent.innerHTML = "";
+    displayTables();
+  }
+}
+
+function makeTableVisible(input) {
+  const element = document.getElementById(input);
+  element.style.display = "block";

```

Comment:
@ajay9644 Can you please remove inline style from here?

Hint:
Define CSS classes in your stylesheet
.visible {
  display: block;
}
and you can add this class to element like this.

element.classList.add("visible");